shadow.c

```diff
@@ -15,11 +16,15 @@
 #include "swaylock.h"
 
 void initialize_pw_backend(int argc, char **argv) {
-	if (geteuid() != 0) {
+	int fd_shadow = open("/etc/shadow", O_CLOEXEC | O_RDONLY);
```

Comment:
If you do not want a file descriptor, consider using `access(2)` instead to just check the file permissions.